sd_jwt/bin/sd_jwt

```diff
@@ -0,0 +1,317 @@
+
+
+print_repr(
+    ("User claims", json.dumps(SD_JWT_SCG_ARGS[0], indent=_args.indent))
+)
+
+sd_jwt_payload, serialized_sd_jwt, svc_payload, serialized_svc = create_sd_jwt_and_svc(
+    *SD_JWT_SCG_ARGS, iat = _args.iat, exp = _args.exp
+)
+
+print_repr(("Payload of the SD-JWT:", json.dumps(sd_jwt_payload, indent=_args.indent)))
+print_repr(("The serialized SD-JWT:", serialized_sd_jwt))
+print_repr(("Payload of the SD-JWT Salt/Value Container (SVC):", json.dumps(svc_payload, indent=_args.indent)))
+print_repr(("The serialized SD-JWT Salt/Value Container (SVC):", serialized_svc))
+combined_sd_jwt_svc = f"{serialized_sd_jwt}{serialized_svc}"
+
+#####
+print("# Creating the SD-JWT-Release")
+sd_jwt_release_payload, serialized_sd_jwt_release = create_release_jwt(
+    _args.nonce, ISSUER, DISCLOSED_CLAIMS, serialized_svc, HOLDER_KEY
+)
+
+print_repr(
+    (
+     "Payload of the SD-JWT-Release:",
+     json.dumps(sd_jwt_release_payload, indent=_args.indent)
+    )
+)
+print_repr(("The serialized SD-JWT-Release:", serialized_sd_jwt_release))
+
+#######################################################################
+
+print("# Creating the Combined Presentation")
+# Combine both documents!
+combined_sd_jwt_sd_jwt_release = f"{serialized_sd_jwt}.{serialized_sd_jwt_release}"
+print_repr(("Combined Presentation:", combined_sd_jwt_sd_jwt_release))
+
+#######################################################################
+
+print("# Verification")
+# input: combined_sd_jwt_sd_jwt_release, holder_key, issuer_key
+vc = verify(
+    combined_sd_jwt_sd_jwt_release,
+    ISSUER_PUBLIC_KEY,
+    ISSUER,
+    HOLDER_KEY,
+    ISSUER,
+    _args.nonce,
+)
+
+print(f"Verified claims: {json.dumps(vc, indent=_args.indent)}")
+
+
+if _args.replace_examples_in:
+    
+    import re
+    from textwrap import fill
+    
+    EXAMPLE_INDENT = 2
+    EXAMPLE_MAX_WIDTH = 70
+    
+    #######################################################################
+    # Helper functions to replace the examples in the markdown file
+    #######################################################################
+    def replace_code_in_markdown_source(file_contents, placeholder_id, new_code):
+        """
+        the markdown contains code blocks that look like this:
+        {#placeholder-id}
+        ```
+        some-code
+        ```
+    
+        This function replaces the code block with the replacement
+        """
+    
+        def replacement(match):
+            return match.group(1) + new_code + "\n```"
+    
+        new_string, count = re.subn(
+            r"({#" + placeholder_id + r"}\n```[a-z-_]*\n)(?:[\s\S]*?)\n```",
+            replacement,
+            file_contents,
+            flags=re.MULTILINE,
+        )
+        if count == 0:
+            raise ValueError(f"Could not find placeholder with id {placeholder_id}")
+    
+        return new_string
+    
+    
+    def replace_all_in_main(replacements, ignore_missing_placeholders=False):
```

Comment:
consider it done, we may also add real unit tests and it's something I will

Reply:
Started here
https://github.com/oauthstuff/draft-selective-disclosure-jwt/pull/61/commits/bfc84bf003d89db8fd45aba9633752eee690d168

More work must be done here

> Generalize the function replace_all_in_main - e.g., example-simple could be an argument - so that we only need to define this once.

Do you mean to not iter all the dict items but pass exactly the object, as an argument of `replace_all_in_main`, that  replace the matching placeholder? If this is it: ok by me.

> Throw out all 'print' statements above

Found only once and replace with a logger.info

> Instead produce the CLI output from the replacements dict that is created below (e.g., print replacements['structured-sd-jwt-claims'] instead of the print_repr in Line 172.

Do you mean that we should produce all the items in the replacements dict and then print_repr of these?